### PR TITLE
[Clang][Driver] Link nsan runtime

### PR DESCRIPTION
**Pull request by alexander-shaposhnikov**
Link nsan runtime.

### CODE REVIEW

**Review by MaskRay — APPROVED** [review]

File: clang/test/Driver/sanitizer-ld.c [hunk 1]
```diff
@@ -627,6 +627,20 @@
...
+// RUN:     --target=x86_64-unknown-linux -fuse-ld=ld -fsanitize=numerical \
+// RUN:     -resource-dir=%S/Inputs/resource_dir \
+// RUN:     --sysroot=%S/Inputs/basic_linux_tree \
+// RUN:   | FileCheck --check-prefix=CHECK-NSAN-LINUX %s
+//
+// CHECK-NSAN-LINUX: "{{(.*[^-.0-9A-Z_a-z])?}}ld{{(.exe)?}}"
+// CHECK-NSAN-LINUX-NOT: "-lc"
+// CHECK-NSAN-LINUX-NOT: libclang_rt.ubsan
+// CHECK-NSAN-LINUX: libclang_rt.nsan.a"
+// CHECK-NSAN-LINUX: "-lpthread"
```
Comment (MaskRay) [inline clang/test/Driver/sanitizer-ld.c:640]: If the three `-l` are consecutive, it's recommended to place them on the same line like `"-lx" "-ly" "-lz"`

File: clang/test/Driver/sanitizer-ld.c [hunk 2]
```diff
@@ -627,6 +627,20 @@
 // CHECK-COV-LINUX: "-lpthread"
 // CHECK-COV-LINUX: "-lresolv"
 
+// RUN: %clang -### %s 2>&1 \
+// RUN:     --target=x86_64-unknown-linux -fuse-ld=ld -fsanitize=numerical \
+// RUN:     -resource-dir=%S/Inputs/resource_dir \
+// RUN:     --sysroot=%S/Inputs/basic_linux_tree \
+// RUN:   | FileCheck --check-prefix=CHECK-NSAN-LINUX %s
+//
+// CHECK-NSAN-LINUX: "{{(.*[^-.0-9A-Z_a-z])?}}ld{{(.exe)?}}"
```
Comment (MaskRay) [inline clang/test/Driver/sanitizer-ld.c:636]: Just `ld{{(.exe)?}}"`

Comment (alexander-shaposhnikov) [reply clang/test/Driver/sanitizer-ld.c]: done

File: clang/test/Driver/sanitizer-ld.c (hunk 1 above)
Comment (alexander-shaposhnikov) [reply clang/test/Driver/sanitizer-ld.c]: done

**Pull request merged**